pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -614,8 +614,8 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 		return nil, fmt.Errorf("cannot allocate unregistered device %s", resource)
 	}
 
-	// Check if registered resource has healthy devices
-	if healthyDevices.Len() == 0 {
+	// Check if registered resource has healthy devices, except if the resource is not required
+	if healthyDevices.Len() == 0 && required != 0 {
```

Comment:
at first glance, we should handle this corner case earlier in the flow, possibly in `allocateContainerResources` before to call this function. And this case probably deserves a log line.

Reply:
The issue arises here. The initial design intended to consider and handle various scenarios here, so it seems appropriate to address it here as well.

Reply:
I agree the issue is raised here, but we should not try to even attempt allocation if we actually have no request. So we can handle this in the upper layer.